File: solutions/python/bob/2/bob.py
import re

def response(hey_bob):
    hey_bob = hey_bob.rstrip()
    if not hey_bob:
        return "Fine. Be that way!"
    is_question = hey_bob.endswith("?")
    is_shout = hey_bob.isupper()
    if is_shout:
        if is_question:
            return "Calm down, I know what I'm doing!"
        return "Whoa, chill out!"
    if is_question:
        return "Sure."
    return "Whatever."

File: solutions/javascript/bob/2/bob.js
export const hey = (message) => {
  const isYelling = /[A-Z]/.test(message) && message === message.toUpperCase();
  const isQuestion = /\?\s*$/.test(message);
  const isSilence = /^\s*$/.test(message);
  
  if (isSilence) return 'Fine. Be that way!';
  if (isQuestion && isYelling) return 'Calm down, I know what I\'m doing!';
  if (isQuestion) return 'Sure.';
  if (isYelling) return 'Whoa, chill out!';
  return 'Whatever.'
}